app/models/user.rb

```diff
@@ -96,6 +100,12 @@ def self.reset_token_expired?(sent_at)
     Time.current > (sent_at.in(RESET_TOKEN_VALIDITY_PERIOD))
   end
 
+  def default_room
+    return unless PermissionsChecker.new(permission_names: 'CreateRoom', user_id: id, current_user: self, friendly_id: nil, record_id: nil).call
+
+    Room.create(name: "#{name}'s Room", user_id: id) if rooms.count <= 0
```

Comment:
Just for performance sake, I would move the `if rooms.count <=0` to the top of the function. That way, we can do a quick check to make sure they don't already have a room